[dist] add reduce_scatter_out

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #168266
* #167647
* __->__ #168260

Adding reduce_scatter_tensor_out to use in fx passes to efficiently decompose reduce_scatter without concatenation.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @chenyang78

IvanKobzarev committed:
[dist] add reduce_scatter_out

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"


Adding reduce_scatter_tensor_out to use in fx passes to efficiently decompose reduce_scatter without concatenation.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

eellison reviewed (commented):
Can you add a lowering in ` comm_lowering.py` as well ?

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"


Adding reduce_scatter_tensor_out to use in fx passes to efficiently decompose reduce_scatter without concatenation.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"


Adding reduce_scatter_tensor_out to use in fx passes to efficiently decompose reduce_scatter without concatenation.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"


Adding reduce_scatter_tensor_out to use in fx passes to efficiently decompose reduce_scatter without concatenation.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"


Adding reduce_scatter_tensor_out to use in fx passes to efficiently decompose reduce_scatter without concatenation.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo chenyang78

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[dist] add reduce_scatter_out"


Adding reduce_scatter_tensor_out to use in fx passes to efficiently decompose reduce_scatter without concatenation.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo chenyang78

[ghstack-poisoned]
